deploying-monitoring-redis-oke.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Deploying and monitoring a Redis cluster to Oracle Container Engine (OKE)
+parent: [tutorials]
+thumbnail: assets/redis-dashboard.png
+tags: [backend, oci]
+date: 2021-12-15 13:34
+description: Redis, Prometheus, OKE, oh my!
+MRM: WWMK211213P00053
+author: ali-mukadam
+toc: false
+---
+In the [previous post](https://medium.com/@lmukadam/extending-terraform-oke-with-a-helm-chart-a51ae0df29d4), we added a simple extension to the [terraform-oci-oke project](https://github.com/oracle-terraform-modules/terraform-oci-oke) so that it uses the [Redis helm chart](https://github.com/helm/charts/tree/master/stable/redis) to deploy a Redis cluster on Kubernetes
+
+In this post, we'll do something more ambitious:
+
+* Deploy a Redis Cluster as in the previous post
+* Monitor the Redis cluster with Prometheus
+* Populate the Redis cluster with existing data using Redis Mass Insertion
+* Visualize the mass insertion process with Grafana
+
+For the sake of convenience, we'll do a manual deployment of Prometheus and Redis. However, if you are using the terraform-oci-oke module (or any Kubernetes cluster for that matter), you can achieve the same result by using the helm provider as described in the previous post.
+
+## Prerequisites
+
+This tutorial requires an account with Oracle Cloud Infrastructure (OCI). If you don't yet have an OCI account, you can quickly sign up for one today by registering for an [Oracle Cloud Free Tier](https://www.oracle.com/cloud/free/#always-free) account. 
+
+Afterwards, check [developer.oracle.com](https://developer.oracle.com) for more developer content.
+
+## Deploy Prometheus Operator
+
+Let's create a namespace for Prometheus:
+
+```console
+kubectl create namespace monitoring
+```
+
+If you are using the terraform-oci-oke module and have provisioned the bastion host, helm is already installed and pre-configured for you. Just login to the bastion and deploy the Prometheus operator:
+
+```console
+helm install --namespace monitoring \
+stable/prometheus-operator \
+--name prom-operator \
+--set kubeDns.enabled=true \
+--set prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues=false \
+--set coreDns.enabled=false \
+--set kubeControllerManager.enabled=false \
+--set kubeEtcd.enabled=false \
+--set kubeScheduler.enabled=false
+```
+
+Setting `serviceMonitorSelectorNilUsesHelmValues` to false ensures that all ServiceMonitors will be selected.
+
+Get a list of pods and identify the Prometheus pods:
+
+<!--EDIT there's a line break missing somewhere between grep and prom- but I'm not sure where-->
```

Comment:
You can remove my HTML comments after editing. Some of these commands appear to be mangled or poorly copy/pasted, please validate against whatever the original source material was.

Reply:
Corrected this and other goofily-formatted blocks